### ISSUE TITLE
Make lf_parse's arguments const and not unsigned

### ISSUE THREAD

**Issue by bkfox** [FoxelSA/libfastcal#1]
`lf_Enum_t lf_parse( lf_Char_t * lfMacAddress, lf_Char_t * lfMountPoint, lf_Descriptor_t * lfDesc )`
to
`lf_Enum_t lf_parse( const char * lfMacAddress, const char* lfMountPoint, lf_Descriptor_t * lfDesc )`

Since lfMacAddress and lfMountPoint are not modified, it should be better to make them const. That will also ease use of cpp's std::string through c_str() (and avoid the need of extra buffer allocation in gnoproj)

**Comment by ghost** [comment]
Thanks for your interest in FOXEL devs.

I'm not very familiar about C/C++ compatibility issues. What exactly would change adding the const qualifier in C++ client software ?

**Comment by bkfox** [comment]
First, `const` helps the user's compiler for optimizations since it has access only to the header file;
Second, if there is no need to use a specific type (e.g. `unsigned char`), the best should be using standard and mostly used ones (like `char`). So, for code and api reference reading, it is easier and cleaner (furthermore "lf_Char_t' to stand as "unsigned char" is confusing because we don't have information about the unsigned). Typedef it's own name around a type when it is not needed implies the reader to find the declaration of the type (here lf_Char_t, in fastcall.h), that results in lost time and complexity.

Also, C++ strings have a c_str() method that returns a const char\* array that is the null-terminated string; This will avoid the code in gnoproj (gnoproj.cpp, from line 146):

```
char *c_mount_point = new char[mount_point.length() + 1];
std::strcpy(c_mount_point, mount_point.c_str());
char *c_mac = new char[mac_address.length() + 1];
std::strcpy(c_mac, mac_address.c_str());
if ( lf_parse( (unsigned char*)c_mac, (unsigned char*)c_mount_point, & lfDesc ) == LF_TRUE ) {
```

This code is memory unsafe (does not handle error on memory allocation and deallocation), not optimized (create new copies), and force an unsafe type conversion. Since this library and gnoproj are created for the same project, I think this might be important to:
- use pure c++ code (since this language includes better memory and error system management);
- think at least of a better compatibility and a cleaner API;

In the case you use `const char*`, the following is possible, that avoid theses previous problems:
`if( lf_parse( mac.c_str(), mountPoint.c_str(), &desc) == LF_TRUE ) {`

Finally, just a little point about LF_TRUE and LF_FALSE: perhaps is it better to return a true boolean with false/true, that is more intuitive about success/fail case if the convention of returning true at success is followed. As a user, I might think that LF_TRUE is `true` and not `1`.

**Comment by bkfox** [comment]
Here is a git diff: that still compile as I have had tested:

``` diff
diff --git a/.gitignore b/.gitignore
deleted file mode 100644
index 9c8bdd2..0000000
--- a/.gitignore
+++ /dev/null
@@ -1,4 +0,0 @@
-bin/
-dev/
-doc/html
-obj/
diff --git a/src/fastcal-parser.c b/src/fastcal-parser.c
index 0564dec..c7c1f8b 100644
--- a/src/fastcal-parser.c
+++ b/src/fastcal-parser.c
@@ -47,7 +47,7 @@
     Source - Key/value-file descriptor management
  */

-    lf_Enum_t lf_parse( lf_Char_t * lfMacAddress, lf_Char_t * lfMountPoint, lf_Descriptor_t * lfDesc ) {
+    lf_Enum_t lf_parse(const lf_Char_t * lfMacAddress, const lf_Char_t * lfMountPoint, lf_Descriptor_t * lfDesc ) {

         /* Path variables */
         lf_Char_t lfPath[256] = { lf_Char_s( 0 ) };
diff --git a/src/fastcal-parser.h b/src/fastcal-parser.h
index 153cc7b..0734b00 100644
--- a/src/fastcal-parser.h
+++ b/src/fastcal-parser.h
@@ -109,7 +109,7 @@
      *  \return Returns key-file descriptor creation status 
      */

-    lf_Enum_t lf_parse ( lf_Char_t * lfMacAddress, lf_Char_t * lfMountPoint, lf_Descriptor_t * lfDesc );
+    lf_Enum_t lf_parse ( const lf_Char_t * lfMacAddress, const lf_Char_t * lfMountPoint, lf_Descriptor_t * lfDesc );

     /*! \brief Key/value-file descriptor management
      *
diff --git a/src/fastcal.h b/src/fastcal.h
index c2263de..85836e4 100644
--- a/src/fastcal.h
+++ b/src/fastcal.h
@@ -155,7 +155,7 @@
  */

     /* Define generic char type */
-    typedef unsigned char lf_Char_t;
+    typedef char lf_Char_t;

     /* Define floating point type */
     typedef double        lf_Real_t;
```

**Comment by ghost** [comment]
Thanks for the advise. Const qualifiers have been added and published in release 1.1.1.

Nevertheless, char will remains unsigned.
